src/main/java/org/folio/rest/validator/CustomPackagePutBodyValidator.java

```diff
@@ -0,0 +1,37 @@
+package org.folio.rest.validator;
+
+import java.util.Optional;
+
+import org.folio.rest.exception.InputValidationException;
+import org.folio.rest.jaxrs.model.Coverage;
+import org.folio.rest.jaxrs.model.PackageDataAttributes;
+import org.folio.rest.jaxrs.model.PackagePutRequest;
+
+public class CustomPackagePutBodyValidator
+{
+  private static final String INVALID_REQUEST_BODY_TITLE = "Invalid request body";
+  private static final String INVALID_REQUEST_BODY_DETAILS = "Json body must contain data.attributes";
+
+  public void validate(PackagePutRequest request) {
+    if (request == null ||
+      request.getData() == null ||
+      request.getData().getAttributes() == null) {
+      throw new InputValidationException(INVALID_REQUEST_BODY_TITLE, INVALID_REQUEST_BODY_DETAILS);
+    }
+    PackageDataAttributes attributes = request.getData().getAttributes();
+    String name = attributes.getName();
+    PackageDataAttributes.ContentType contentType = attributes.getContentType();
+    String beginCoverage = Optional.ofNullable(attributes.getCustomCoverage())
+      .map(Coverage::getBeginCoverage)
+      .orElse(null);
+
+    String endCoverage = Optional.ofNullable(attributes.getCustomCoverage())
+      .map(Coverage::getEndCoverage)
+      .orElse(null);
+
+    ValidatorUtil.checkIsNotEmpty("name", name);
+    ValidatorUtil.checkIsNotNull("contentType", contentType);
```

Comment:
Do we also want to check the validity of the content type? We may not have been doing this in the ruby code per below --but there was a note for checking it

https://github.com/folio-org/mod-kb-ebsco/blob/9b68934c587983e28cb7a3d9c73e7a562b31ac90/app/controllers/validation/custom_package_parameters.rb#L36

Reply:
ContentType is enum, so this validation is automatically handled by RMB module

Reply:
Ok thanks for clarification @andrii-paias